jspoon/src/test/java/pl/droidsonroids/jspoon/AdvancedTest.java

```diff
@@ -0,0 +1,198 @@
+        @Selector(value = "#quote",
+                  format = "\"(.*)\"") String quote;
+        List<AnnotatedPost> posts;
+    }
+
+    private static class PageLFieldWithoutAnnotation {
+        @Selector(value = "#quote",
+                  format = "\"(.*)\"") String quote;
+        AnnotatedPost firstPost;
+    }
+
+    private static class Post {
+        @Selector(value = ":root",
+                  attr = "data-created",
+                  format = "HH:mm dd.MM.yyyy") Date created;
+        @Selector("h2") String header;
+        @Selector("p") String content;
+        @Selector("li") List<String> tags;
+
+        Post() {
+            //no-op
+        }
+
+        Post(Date created, String header, String content, String... tags) {
+            this.created = created;
+            this.header = header;
+            this.content = content;
+            this.tags = Arrays.asList(tags);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            Post post = (Post) o;
+
+            if (created != null ? !created.equals(post.created) : post.created != null) return false;
+            if (header != null ? !header.equals(post.header) : post.header != null) return false;
+            if (content != null ? !content.equals(post.content) : post.content != null) return false;
+            return tags != null ? tags.equals(post.tags) : post.tags == null;
+        }
+
+        @Override
+        public int hashCode() {
+            return 0;
+        }
+    }
+
+    @Selector(".post")
+    private static class AnnotatedPost extends Post {
+        @Selector(value = ":root",
+                  attr = "data-created",
+                  format = "HH:mm dd.MM.yyyy") Date created;
+        @Selector("h2") String header;
+        @Selector("p") String content;
+        @Selector("li") List<String> tags;
+
+        AnnotatedPost() {
+            //no-op
+        }
+
+        AnnotatedPost(Date created, String header, String content, String... tags) {
+            this.created = created;
+            this.header = header;
+            this.content = content;
+            this.tags = Arrays.asList(tags);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            AnnotatedPost post = (AnnotatedPost) o;
+
+            if (created != null ? !created.equals(post.created) : post.created != null) return false;
+            if (header != null ? !header.equals(post.header) : post.header != null) return false;
+            if (content != null ? !content.equals(post.content) : post.content != null) return false;
+            return tags != null ? tags.equals(post.tags) : post.tags == null;
+        }
+
+        @Override
+        public int hashCode() {
```

Comment:
hashCode jest niespójne z equals.
https://stackoverflow.com/questions/1678205/in-java-why-must-equals-and-hashcode-be-consistent